### PR TITLE
Add new FileIcon distinction between List and ListItem

### PR DESCRIPTION
**Pull request by bigbadcapers**
#### Pull request checklist

- [x] Addresses an existing issue: Update List-related FileType icons to new brand
- [x] Create clearer distinction in the FileTypeIcon enum between List and ListItem
- [x] Include a change request file using `$ yarn change`

#### Focus areas to test

getFileTypeIconProps method with FileIconType.listItem and FileIconType.list as submitted types.

### CODE REVIEW

File: packages/file-type-icons/src/FileIconType.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ export enum FileIconType {
   documentsFolder = 10,
   picturesFolder = 11,
   linkedFolder = 12,
+  list = 13,
 }
 
-export type FileIconTypeInput = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+export type FileIconTypeInput = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13;
```
Comment (ThomasMichon) [inline packages/file-type-icons/src/FileIconType.ts:24]: Please change this to `keyof typeof FileIconType`.

Comment (bigbadcapers) [reply packages/file-type-icons/src/FileIconType.ts:24]: replace the entire line with that? this gave me a compile error until i added the 13, and it seems related to this test case but im frankly not sure what its other purpose is

https://github.com/microsoft/fluentui/blob/ee40f3275316d43f9ec10ec3cf11ed9c4d3e64f9/packages/file-type-icons/src/FileIconType.test.ts#L5

**Review by ThomasMichon — APPROVED** [review]

**Review by ecraig12345 — CHANGES_REQUESTED** [review]
Thanks for submitting this fix, but due to work we're currently doing to prepare `master` for our version 8 beta release, we're asking contributors to either wait a couple weeks to submit fixes (if it's not urgent) or submit to the new `7.0` branch (if it's urgent). See #15222 for more details, and sorry about the inconvenience.

**Comment by bigbadcapers** [comment]
> Thanks for submitting this fix, but due to work we're currently doing to prepare `master` for our version 8 beta release, we're asking contributors to either wait a couple weeks to submit fixes (if it's not urgent) or submit to the new `7.0` branch (if it's urgent). See #15222 for more details, and sorry about the inconvenience.

Ok, I switched the PR's target and it seems like it's working through the check now alright without having resubmitted it. Please lmk if I need to do something different, thanks!

**Comment by bigbadcapers** [comment]
@Jahnp can you help to merge this one?

**Review by Jahnp — APPROVED** [review]

**Comment by Jahnp** [comment]
@bigbadcapers sure thing, thanks.

@ecraig12345 , I know you requested changes on the branch target, but it looks like that’s been fixed, so I’m going to go ahead and merge since I believe this is pretty urgent.

**Comment by ecraig12345** [comment]
@Jahnp Yup I just forgot to update my review, so thanks for taking care of this!

**Pull request merged**